### ISSUE TITLE
Segfault:  TorchConverter derived program segfaults on .optimize()

### ISSUE THREAD

**Issue by andremolnar** [apple/coreai-torch#33]
MacOS 27ß2 - coreai-torch 0.4.1

to_coreai()	succeeds
save_asset()	succeeds (produces a valid .aimodel)
optimize()	segfault
Individual passes (legalize-to-core, core-to-odix)	segfault

Minimal script to reproduce

```python
# /// script
# requires-python = ">=3.11"
# dependencies = [
#     "coreai-torch==0.4.1",
#     "coreai-core==1.0.0b2",
#     "torch==2.11.0",
#     "transformers>=4.44",
#     "einops>=0.7",
#     "kornia>=0.7",
#     "timm>=1.0",
# ]
# ///
import torch
from transformers import AutoModelForImageSegmentation

model = AutoModelForImageSegmentation.from_pretrained(
    "ZhengPeng7/BiRefNet_HR-matting",
    trust_remote_code=True,
    torch_dtype=torch.float32,
).eval()

class Wrapper(torch.nn.Module):
    def __init__(self, m):
        super().__init__()
        self.model = m
        self.register_buffer("mean", torch.tensor([0.485,0.456,0.406]).view(1,3,1,1))
        self.register_buffer("std",  torch.tensor([0.229,0.224,0.225]).view(1,3,1,1))
    def forward(self, x):
        x = (x - self.mean) / self.std
        return torch.sigmoid(self.model(x)[-1])

w = Wrapper(model).eval()
ep = torch.export.export(w, args=(torch.randn(1,3,256,256),))
from coreai_torch import TorchConverter, get_decomp_table
ep = ep.run_decompositions(get_decomp_table())
program = TorchConverter().add_exported_program(
    ep, input_names=["image"], output_names=["alpha"]
).to_coreai()
program.optimize()  # SEGFAULT in apply_passes_sync
```

Key details for the issue:
Item	Value
coreai-torch	0.4.1 (also tried 0.4.0)
coreai-core	1.0.0b2
torch	2.11.0
Model	BiRefNet Swin-Large (220M params), ZhengPeng7/BiRefNet_HR-matting
Unique aten ops in graph	38 (standard set: conv2d, matmul, softmax, layer_norm, roll, pad, gelu, upsample_bilinear2d, etc.)
to_coreai()	succeeds
save_asset()	succeeds (produces valid 426 MB .aimodel at fp16)
optimize()	segfault
Individual passes (legalize-to-core, core-to-odix)	segfault

**Comment by andremolnar** [comment]
Tested again in Mac OS 27.0 Beta 3 (26A5378n) and I am no longer seeing a crash (all other things being equal).

**Comment by jakesabathia2** [comment]
Thanks for the update. If you start seeing the issue, please re-open this issue.
